k8s_snapshots/snapshot.py

```diff
@@ -365,9 +365,13 @@ def determine_next_snapshot(snapshots, rules):
     for rule in rules:
         _log = _logger.new(rule=rule)
         # Find all the snapshots that match this rule
+        # This returns a <filter> object
         snapshots_for_rule = filter_snapshots_by_rule(snapshots, rule)
-        # Rewrite the list to snapshot
-        snapshot_times = map(lambda s: s.created_at, snapshots_for_rule)
```

Comment:
aw, this map() call was nice! it appears your loop below does the same thing

Reply:
(although a [list comprehension](https://docs.python.org/3.6/tutorial/datastructures.html#list-comprehensions) would be more pythonic)

Reply:
Unfortunately, the map call was leading to unexpected behaviour in that it was not populating the variable correctly before, and now it does. Jay can attest to this.

Reply:
oh, since it was returning a lazy iterator instead of a list. ok. a list comprehension would be a bit cleaner, but this makes sense now